crates/router/src/connector/nmi.rs

```diff
@@ -325,12 +342,14 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
         req: &types::PaymentsAuthorizeRouterData,
         _connectors: &settings::Connectors,
     ) -> CustomResult<RequestContent, errors::ConnectorError> {
-        let connector_router_data = nmi::NmiRouterData::try_from((
-            &self.get_currency_unit(),
+        let amount = connector_utils::convert_amount(
+            self.amount_converter,
+            req.request
+                .test_amount
+                .unwrap_or(MinorUnit::new(req.request.amount)),
```

Comment:
If `test_amount` is present, this will convert it and send to the connector regardless of the requested amount. This is opening a lot of scope for wrong amount being sent to the connector

Reply:
correct, thinking of some other strategy at this point, will convert the amount  in connector module to MInorUnit and do the conversion, once it is done for all the connectors will change it in parent data!